### PR TITLE
feat(components/forms): add help inline popover to file attachment

### PR DESCRIPTION
**Pull request by Blackbaud-CoreyArcher**
[AB#2835418](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2835418)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2263/storybook/)

Component Storybooks:

- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/colorpicker-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/errors-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/forms-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/lookup-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/phone-field-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/2263/storybooks/theme-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2263/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/2263/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2263/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2263/playground/)

File: libs/components/forms/src/lib/modules/file-attachment/file-drop.component.ts [hunk 1]
```diff
@@ -178,6 +179,23 @@ export class SkyFileDropComponent implements OnInit, OnDestroy {
   @Input({ transform: booleanAttribute })
   public required = false;
 
+  /**
+   * The content of the help popover. When specified along with `labelText`, a [help inline](https://developer.blackbaud.com/skyux/components/help-inline)
+   * button is added to date range picker. The help inline button displays a [popover](https://developer.blackbaud.com/skyux/components/popover)
```
Comment (Blackbaud-SandhyaRajasabeson) [inline libs/components/forms/src/lib/modules/file-attachment/file-drop.component.ts]: should be `file attachment` not `date range picker`

File: apps/e2e/forms-storybook/src/app/file-attachment/file-attachment.component.html [hunk 2]
```diff
@@ -30,6 +32,7 @@
 
 <div style="width: 60%" *ngIf="basic">
   <sky-file-drop
+    helpPopoverTitle="This should not show up"
     labelText="File attachment links removed"
     [acceptedTypes]="acceptedTypes"
     [allowLinks]="false"
```
Comment (Blackbaud-SandhyaRajasabeson) [inline apps/e2e/forms-storybook/src/app/file-attachment/file-attachment.component.html:39]: missing scenarios: 
- easy mode file attachment without help inline
- easy mode file attachment with required asterisk and help inline

File: libs/components/forms/src/lib/modules/file-attachment/file-drop.component.ts (hunk 1 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/forms/src/lib/modules/file-attachment/file-drop.component.ts]: fixed

File: apps/e2e/forms-storybook/src/app/file-attachment/file-attachment.component.html (hunk 2 above)
Comment (Blackbaud-CoreyArcher) [reply apps/e2e/forms-storybook/src/app/file-attachment/file-attachment.component.html:39]: On the `basic` screenshot there's a third file attachment without help inline. Added required to the `no-preview` screenshot.

**Review by Blackbaud-SandhyaRajasabeson — APPROVED** [review]

**Pull request merged**